Makefile

```diff
@@ -8,13 +8,16 @@ ECHO1 = $(V:1=@ :)
 ECHO = $(ECHO1:0=@ echo)
 FUZZ_OUTPUT_DIR = $(shell pwd)/fuzz/output
 
-SOEXT := $(shell ruby -e 'puts RbConfig::CONFIG["SOEXT"]')
+SOEXT ?= $(shell ruby -e 'puts RbConfig::CONFIG["SOEXT"]')
 
 CPPFLAGS := -Iinclude $(CPPFLAGS)
 CFLAGS := -g -O2 -std=c99 -Wall -Werror -Wextra -Wpedantic -Wundef -Wconversion -Wno-missing-braces -fPIC -fvisibility=hidden $(CFLAGS)
-CC := cc
+CC ?= cc
 WASI_SDK_PATH := /opt/wasi-sdk
 
+MAKEDIRS ?= mkdir -p
+RMALL ?= rm -f -r
+
```

Comment:
Of course, `MAKEDIRS` and `RMALL` are redundant since it’s the same for all of *nix and MSYS.
But if [`mkmf` supports challengers who venture without popular gear](https://github.com/ruby/ruby/blob/v3_3_0/lib/mkmf.rb#L2092-L2099), so should we.